maskpostgresdata/management/commands/dump_masked_data.py

```diff
@@ -49,86 +50,91 @@ def reset_sequences(self, cursor):
-        if passwd:
-            subprocess_env["PGPASSWORD"] = str(passwd)
-
-        masker_args = getattr(settings, "MASKER_ARGS", ["--no-owner", "--no-privileges"])
-        if masker_args:
-            args += masker_args
-
-        connection.ensure_connection()
-        connection.set_autocommit(False)
-
-        if connection.isolation_level != ISOLATION_LEVEL_SERIALIZABLE:
-            connection.connection.set_session(isolation_level=ISOLATION_LEVEL_SERIALIZABLE)
-            connection.isolation_level = ISOLATION_LEVEL_SERIALIZABLE
-            connection.connection.isolation_level = ISOLATION_LEVEL_SERIALIZABLE
-
-        cursor = connection.cursor()
-        cursor.execute("SELECT pg_export_snapshot();")
-        snapshot_id = cursor.fetchone()[0]
-
-        args += ["--snapshot={}".format(snapshot_id)]
-
-        header_dump = args + ["--section=pre-data"]
-        subprocess.run(header_dump, stdout=self.stdout._out, env=subprocess_env)
-
-        fields_to_mask = getattr(settings, "MASKER_FIELDS", None)
-        altered_tables = []
-
-        for app in apps.get_app_configs():
-            for model in app.get_models():
-                table_name = model._default_manager.model._meta.db_table
-                if hasattr(self, 'update_{}'.format(table_name)):
-                    getattr(self, 'update_{}'.format(table_name))(model._default_manager.all())
-
-        if fields_to_mask:
-            for app in fields_to_mask.keys():
-                for model, fields in fields_to_mask[app].items():
-                    model_class = apps.get_model(app.lower(), model_name=model.lower())
-                    model_class._default_manager.update(**fields)
-                    table_name = model_class._default_manager.model._meta.db_table
-
-                    altered_tables.append(table_name)
-                    print("COPY public.{} FROM stdin;".format(table_name), flush=True)
-                    cursor.copy_to(self.stdout._out, table_name)
-                    print("\\.\n", file=self.stdout._out, flush=True)
-
-        for app in apps.get_app_configs():
-            for model in app.get_models():
-                table_name = model._default_manager.model._meta.db_table
-
-                if table_name not in altered_tables:
-                    print("COPY public.{} FROM stdin;".format(table_name), flush=True)
-                    cursor.copy_to(self.stdout._out, table_name)
-                    print("\\.\n", file=self.stdout._out, flush=True)
-
-        print("COPY public.django_migrations FROM stdin;".format(table_name), flush=True)
-        cursor.copy_to(self.stdout._out, 'django_migrations')
-        print("\\.\n", file=self.stdout._out, flush=True)
-
-        # Sets a new values for sequences.
-        self.reset_sequences(cursor)
-
-        post_data_dump = args + ["--section=post-data"]
-        subprocess.run(post_data_dump, stdout=self.stdout._out, env=subprocess_env)
-
-        transaction.rollback()
+        try:
```

Comment:
It would be much better to put all this into a separate method / function, and wrap calling that in the `try`/`except`.  Putting the whole body into a massive `try` is asking for pain.